Applications/Spire/Source/Ui/OverlayPanel.cpp

```diff
@@ -45,6 +45,55 @@ namespace {
   }
 }
 
+class OverlayPanel::ParentMoveObserver : public QObject {
+  public:
+    using MoveSignal = Signal<void()>;
+
+    ParentMoveObserver(QWidget& widget) {
+      install_parent_event_filter(widget);
+    }
+
+    ~ParentMoveObserver() {
+      for(auto parent : m_parents) {
+        parent->removeEventFilter(this);
+      }
+    }
+
+    connection connect_move_signal(const MoveSignal::slot_type& slot) const {
+      return m_move_signal.connect(slot);
+    }
+
+    bool eventFilter(QObject* watched, QEvent* event) override {
+      if(event->type() == QEvent::Move) {
+        m_move_signal();
```

Comment:
Is it possible that a single move could cause a move event to be sent to multiple parents, or is it always the case that only an immediate widget is affected by a move?

Reply:
Refer to Qt documentation: “Move events are sent to widgets that have been moved to a new position relative to their parent.”
https://doc.qt.io/qt-5/qmoveevent.html#details

It means that only when the position relative to the parent changes, the move event is emitted. The parent moves, hence the rect of the widget moves, but children's rects are still left exactly the same relative to their parent's rect, so they aren’t moving, hence no move event.

However, it is possible that multiple move events are sent from different widgets when a widget is moving. For example, a parent widget moves and one of its children moves to a new position relative to the parent widget for some reason, the parent and the child respectively trigger move events, but these two move events are different.